src/main/scala/vectorpipe/VectorPipe.scala

```diff
@@ -174,26 +176,22 @@ object VectorPipe {
     *
     * @see [[vectorpipe.Collate]]
     */
-  def toVectorTile[G <: Geometry, D](
-    collate: (Extent, Iterable[Feature[G, D]]) => VectorTile,
-    ld: LayoutDefinition,
-    rdd: RDD[(SpatialKey, Iterable[Feature[G, D]])]
-  ): RDD[(SpatialKey, VectorTile)] = {
+  def toVectorTile[G <: Geometry, D](ld: LayoutDefinition, rdd: RDD[(SpatialKey, Iterable[Feature[G, D]])])
+                  (collate: (Extent, Iterable[Feature[G, D]]) => VectorTile): RDD[(SpatialKey, VectorTile)] = {
     val mt: MapKeyTransform = ld.mapTransform
 
     rdd.map({ case (k, iter) => (k, collate(mt(k), iter))})
   }
 
-  private def logString[G <: Geometry, D](e: Extent, f: Feature[G, D]): String =
-    s"CLIP FAILURE W/ EXTENT: ${e}\nELEMENT METADATA: ${f.data}\nGEOM: ${f.geom.reproject(WebMercator, LatLng).toGeoJson}"
 
-  /** Print any clipping errors to STDOUT - to be passed to [[toGrid]]. */
-  def stdout[G <: Geometry, D](e: Extent, f: Feature[G, D]): Unit = println(logString(e, f))
 
-  /** Log a clipping error as an ERROR through Spark's default log4j - to be passed to [[toGrid]]. */
-  def log4j[G <: Geometry, D](e: Extent, f: Feature[G, D]): Unit =
-    Logger.getRootLogger().error(logString(e, f))
+  /** Log an error to STDOUT. */
+  def logToStdout[A](f: A => String): A => Unit = { a => println(f(a)) }
+
+  /** Log an error as an ERROR through Spark's default log4j. */
+  def logToLog4j[A](f: A => String): A => Unit = { a => Logger.getRootLogger().error(f(a)) }
+
+  /** Skip over some failure. */
+  def logNothing[A](f: A => String): A => Unit = { _ => Unit }
```

Comment:
Notice both the `f` and `a` are in scope. One's custom logging function could then choose to ignore whatever default `f` had been passed up from the depths of VectorPipe and render the error in whatever way they want.

Reply:
I like these example implementations